src/main/java/net/dv8tion/jda/api/interactions/commands/build/EntryPointCommandData.java

```diff
@@ -0,0 +1,217 @@
+import net.dv8tion.jda.api.interactions.commands.DefaultMemberPermissions;
+import net.dv8tion.jda.api.interactions.commands.build.attributes.*;
+import net.dv8tion.jda.api.interactions.commands.localization.LocalizationFunction;
+import net.dv8tion.jda.api.requests.restaction.GlobalCommandListUpdateAction;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.api.utils.data.SerializableData;
+import net.dv8tion.jda.internal.interactions.EntryPointCommandDataImpl;
+import net.dv8tion.jda.internal.utils.Checks;
+import net.dv8tion.jda.internal.utils.localization.LocalizationUtils;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+
+/**
+ * Builder for Entry Point Commands.
+ * <br>Use {@link Commands#entryPoint(String, String)} to create instances of this interface.
+ *
+ * <p>This command can only be added via {@link GlobalCommandListUpdateAction#setEntryPointCommand(EntryPointCommandData)}.
+ *
+ * @see Commands#entryPoint(String, String)
+ */
+public interface EntryPointCommandData
+        extends IDescribedCommandData, INamedCommandData, IScopedCommandData, IRestrictedCommandData,
+        IAgeRestrictedCommandData, SerializableData
+{
+    @Nonnull
+    @Override
+    EntryPointCommandData setLocalizationFunction(@Nonnull LocalizationFunction localizationFunction);
+
+    @Nonnull
+    @Override
+    EntryPointCommandData setName(@Nonnull String name);
+
+    @Nonnull
+    @Override
+    EntryPointCommandData setNameLocalization(@Nonnull DiscordLocale locale, @Nonnull String name);
+
+    @Nonnull
+    @Override
+    EntryPointCommandData setNameLocalizations(@Nonnull Map<DiscordLocale, String> map);
+
+    @Nonnull
+    @Override
+    EntryPointCommandData setDescription(@Nonnull String description);
+
+    @Nonnull
+    @Override
+    EntryPointCommandData setDescriptionLocalization(@Nonnull DiscordLocale locale, @Nonnull String description);
+
+    @Nonnull
+    @Override
+    EntryPointCommandData setDescriptionLocalizations(@Nonnull Map<DiscordLocale, String> map);
+
+    @Nonnull
+    @Override
+    EntryPointCommandData setDefaultPermissions(@Nonnull DefaultMemberPermissions permission);
+
+    @Nonnull
+    @Override
+    EntryPointCommandData setGuildOnly(boolean guildOnly);
+
+    @Nonnull
+    @Override
+    EntryPointCommandData setNSFW(boolean nsfw);
+
+    /**
+     * Sets the handler of this entry point command.
+     * This defines the behavior when this command is used.
+     *
+     * @param  handler
+     *         The handler type for this command
+     *
+     * @return This builder instance, for chaining
+     */
+    @Nonnull
+    EntryPointCommandData setHandler(@Nonnull Handler handler);
+
+    /**
+     * Converts the provided {@link Command} into a {@link EntryPointCommandData} instance.
+     *
+     * @param  command
+     *         The command to convert
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided or the command has illegal configuration
```

Comment:
call out specifically in the docs that this only works for Commands that are of PRIMARY_ENTRY_POINT type